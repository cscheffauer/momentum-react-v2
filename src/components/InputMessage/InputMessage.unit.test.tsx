import InputMessage from '.';
import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';

describe('InputMessage', () => {
  const mountAndWait = async (component) => {
    const _container = mount(component);
    await waitForComponentToPaint(_container);
    return _container;
  };

  const waitForComponentToPaint = async (wrapper: ReactWrapper) => {
    await act(async () => {
      await new Promise((resolve) => setTimeout(resolve, 0));
      wrapper.update();
    });
  };

  describe('snapshot', () => {
    it('should match snapshot', async () => {
      const container = await mountAndWait(<InputMessage message="some message" level="error" />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should render warning triangle for an error', async () => {
      const component = (
        await mountAndWait(<InputMessage message="some message" level="error" />)
      ).childAt(0);
      expect(component.find('mdc-icon').length).toBe(1);
    });

    it('should render warning triangle for a warning', async () => {
      const component = (
        await mountAndWait(<InputMessage message="some message" level="warning" />)
      ).childAt(0);
      expect(component.find('mdc-icon').length).toBe(1);
    });

    it('should not render an icon (right now anyway) for the success case', async () => {
      const component = (
        await mountAndWait(<InputMessage message="some message" level="success" />)
      ).childAt(0);
      expect(component.find('mdc-icon').length).toBe(0);
    });

    it('should not render an icon for the no level case', async () => {
      const component = (
        await mountAndWait(<InputMessage message="some message" level="none" />)
      ).childAt(0);
      expect(component.find('mdc-icon').length).toBe(0);
    });

    it('should have custom class if provided', async () => {
      const testClass = 'testClass';

      const wrapper = await mountAndWait(
        <InputMessage message="some message" className={testClass} />
      );
      const element = wrapper.find(InputMessage).getDOMNode();

      expect(element.classList.contains(testClass)).toBe(true);
    });

    it('should have id if provided', async () => {
      const testId = 'testId';

      const wrapper = await mountAndWait(<InputMessage message="some message" id={testId} />);
      const element = wrapper.find(InputMessage).getDOMNode();

      expect(element.id).toEqual(testId);
    });
  });
});
