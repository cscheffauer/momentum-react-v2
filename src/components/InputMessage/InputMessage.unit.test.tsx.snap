// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputMessage snapshot should match snapshot 1`] = `
<InputMessage
  level="error"
  message="some message"
>
  <div
    className="md-input-message-wrapper"
  >
    <div
      className="md-input-message-message"
      message-level="error"
    >
      <div
        className="md-input-message--icon"
      >
        <Icon
          fillColor="var(--mds-color-theme-text-error-normal)"
          name="warning"
          scale={16}
          weight="bold"
        >
          <Icon
            className="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink"
            data-autoscale={false}
            data-scale={16}
            data-test="warning"
            name="warning-bold"
            style={
              Object {
                "--mdc-icon-fill-color": "var(--mds-color-theme-text-error-normal)",
              }
            }
          >
            <mdc-icon
              class="md-icon-wrapper md-icon-auto-scales md-icon-scales md-icon-no-shrink"
              data-autoscale={false}
              data-scale={16}
              data-test="warning"
              style={
                Object {
                  "--mdc-icon-fill-color": "var(--mds-color-theme-text-error-normal)",
                }
              }
              suppressHydrationWarning={true}
            />
          </Icon>
        </Icon>
      </div>
      <Text
        className="md-input-message--text"
        tagName="small"
        type="body-secondary"
      >
        <Text
          className="md-text-wrapper md-input-message--text"
          tagname="small"
          type="body-midsize-regular"
        >
          <mdc-text
            class="md-text-wrapper md-input-message--text"
            suppressHydrationWarning={true}
          >
            some message
          </mdc-text>
        </Text>
      </Text>
    </div>
  </div>
</InputMessage>
`;
